vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/SortingIT.java

```diff
@@ -61,6 +62,57 @@ public void setInitialSortOrder_changeOrderFromServer_sortIndicatorsUpdated() {
         findElement(By.id("sort-by-age")).click();
         assertAscendingSorter("Age");
     }
+    @Test
+    public void keepSortStatesAfterReAttach() {
+        WebElement btnAttach = findElement(By.id("btn-attach"));
+        WebElement btnRemove = findElement(By.id("btn-detach"));
+        GridElement sortingGridElement = $(GridElement.class).first();
+        findElement(By.id("sort-by-age")).click();
+        // Attach grid
+        btnAttach.click();
+
+        WebElement sortingGrid = findElement(By.id("sorting-grid"));
+        Assert.assertEquals("asc", sortingGrid.findElements(By.tagName("vaadin-grid-sorter"))
+                .get(1).getAttribute("direction"));
+        String textAgeColumnBeforeReattch = sortingGridElement.getCell(0, 1).getText();
+        // Detach
+        btnRemove.click();
+        // Reattach
+        btnAttach.click();
+        sortingGridElement = $(GridElement.class).first();
+        sortingGrid = findElement(By.id("sorting-grid"));
+        Assert.assertEquals("asc", sortingGrid.findElements(By.tagName("vaadin-grid-sorter"))
+                .get(1).getAttribute("direction"));
+        String textAgeColumnAfterReattch = sortingGridElement.getCell(0, 1).getText();
+        Assert.assertEquals(textAgeColumnBeforeReattch, textAgeColumnAfterReattch);
+    }
+
+    @Test
+    public void indicatorsSortStateNumbers() {
+        WebElement btnAttach = findElement(By.id("btn-attach"));
+        WebElement btnRemove = findElement(By.id("btn-detach"));
+
+        // Attach grid
+        btnAttach.click();
+        findElement(By.id("sort-by-age")).click();
+        WebElement sortingGrid = findElement(By.id("sorting-grid"));
+        Assert.assertEquals("asc", sortingGrid.findElements(By.tagName("vaadin-grid-sorter"))
+                .get(1).getAttribute("direction"));
+
+        String sortStateNumberAgeColumn
+                = sortingGrid.findElements(By.tagName("vaadin-grid-sorter")).get(1).getAttribute("_order");
```

Comment:
Sort order isn't used/tested in this test case. Just one column is sorted so `_order` is always `null`.

Reply:
I changed the test, and the grid is now multiSort. I check 2 columns are sorted